src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+
+namespace Windows.Storage
+{
+	public partial class FileIO
+	{
+
+		private static async Task AppendWriteTextAsync(global::Windows.Storage.IStorageFile file, string contents, global::Windows.Storage.Streams.UnicodeEncoding encoding, bool append)
+		{
+			if (file is null)
```

Comment:
```suggestion
			`file = file ?? throw new NullReferenceException("StorageFile cannot be null");`
```

Reply:
is this more readable?

Reply:
I would suggest throwing a `ArgumentNullException` here instead, as it is clearer to the user, that it is not a bug, but in fact a usage error.

Reply:
But UWP sends NullReferenceException, at least on my PC :)

Reply:
Oh, in that case their mistake 😀 . Thanks for clarification 👍

Reply:
My apologies I missed that one, we should never throw a NullReferenceException. This is most probably an invalid validation on the WinRT side, let's throw a ArgumentNullException instead. Thanks @MartinZikmund .

Reply:
I checked and used same exception as UWP - to be even "bug compatible".
When someone wrote some app, with try/catch and check which exception is thrown, he would expect and handle NullReferenceException, not ArgumentNullException. When Uno would throw another exception, catch{} would be something like `if( (Uno && ArgumentNullException ) || (!Uno && NullReferenceException)) { handleBadFile() }` ...
Is this what we want?

Reply:
In either case, it's a bad practice to handle such errors that way. Throwing a `NullReferenceException` is misleading because it's not an actual NRE. Let's throw an `ArgumentNullException` as it will show in the debugger more interesting details.